smsgateway/backends/redistore.py

```diff
@@ -45,13 +46,15 @@ def _initialize(self, sms_request, account_dict):
                                                password=account_dict['pwd'])
         return True
 
-
     def _get_sms_list(self, sms_request):
         if not sms_request:
             return []
         sms_list = []
-        self.reference = (datetime.datetime.now().strftime('%Y%m%d%H%M%S')
-                          + '+' + u''.join(sms_request.to[:1]))
+        self.reference = (
+            str(uuid.uuid4()) +
+            '+' +
+            u''.join(sms_request.to[:1])
+        )
```

Comment:
@noxxan this could get quite dangerous.
Maybe better to concat `datetime.datetime.now().strftime('%Y%m%d%H%M%S') + sms_request.to[:1] + sms_request.msg`.

Then we can send different messages to the same user in given second, looks much safer to me.